### PR TITLE
feat: decouple retrieval results from retrievals

### PR DESCRIPTION
**Pull request by bajtos**
1. Decouple `retrieval_results` table from `retrieval` to allow us to submit results for retrieval tasks with no linked `retrieval` row.

2. Rename `retrieval_results` to `measurements` to align with MERidian vocabulary.

3. Update `setRetrievalResult` to support the new data schema. This allows existing SPARK checkers to keep reporting measurements.

4. Remove `GET /retrieval/:id` endpoint and replace it with a new endpoint `GET /measurements/:id`. This endpoint is used by integration tests in SPARK checker, we will need to update these tests after this PR lands.

5. Create a new endpoint for submitting retrieval results based on per-round tasks - `POST /measurements`.

I am envisioning the following workflow for the SPARK checker node:
1. Get the current round & the tasks via `GET /rounds/current` - see #80
2. Pick a task, perform the retrieval
3. Call the new API `POST /measurement` to submit the result
4. Rinse and repeat

Connect to https://github.com/filecoin-station/spark/issues/13

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -475,16 +451,16 @@ describe('Routes', () => {
...
       await assertResponseStatus(res, 200)
       const body = await res.json()
-      assert.strictEqual(body.id, retrievalId)
+      assert.strictEqual(body.id, retrievalResultId)
       assert.strictEqual(body.cid, cid)
       assert.strictEqual(body.providerAddress, providerAddress)
       assert.strictEqual(body.protocol, protocol)
       assert.strictEqual(body.sparkVersion, '1.2.3')
       assert.strictEqual(body.zinniaVersion, '2.3.4')
-      assert(body.createdAt)
```
Comment (bajtos) [inline test/test.js:487]: In the new data schema, we don't have "retrieval created at", as this information won't be available in the new SPARK checker version.

We can use `finishedAt` field instead - this requires updating all Grafana charts.

Thoughts?

File: test/test.js [hunk 2]
```diff
@@ -356,7 +358,10 @@ describe('Routes', () => {
       )
       await assertResponseStatus(res, 200)
     })
-    it('ignores duplicate submissions', async () => {
+
+    // Duplicate submissions are ok, because we filter out duplicates in the fraud detection step
+    it('allows duplicate submissions', async () => {
```
Comment (bajtos) [inline test/test.js:363]: Do we have any concerns about DoS attacks? Before my change, clients could submit only as many results as many measurements they created. After my change, they can remove as many measurements as they like. I think my change is not making the situation significantly worse than it was before.

**Comment by juliangruber** [comment]
> Decouple retrieval_results table from retrieval to allow us to submit results for retrieval tasks with no linked retrieval row

I wonder if we even still need the `retrievals` table, now that we have `retrieval_tasks`. Would it make sense to merge `retrievals` into `retrieval_tasks`?

**Comment by juliangruber** [comment]
> Remove GET /retrieval/:id endpoint and replace it with a new endpoint GET /retrieval-results/:id. These endpoints are used by integration tests in SPARK checker, we will need to update these tests after this PR lands.

Ideally (I think) we'll keep `/retrievals/:id` and make that serve retrieval tasks 🤔

**Comment by bajtos** [comment]
@juliangruber I updated the PR based on our IRL discussion:

- Renamed `retrieval_results` to `measurements`
- Renamed `GET /retrieval-results/:id` to `GET /measurements/:id`

I also added a new endpoint `POST /measurements` to be used by the updated SPARK module.

I think this PR is going to break our Grafana charts, I'll update the queries to use the new table name `measurements` after this PR is landed.

File: test/test.js (hunk 1 above)
Comment (juliangruber) [reply test/test.js:487]: Sgtm!

File: test/test.js (hunk 2 above)
Comment (juliangruber) [reply test/test.js:364]: What do you mean by "clients can remove as many measurements as they like"?

File: index.js [hunk 3]
```diff
@@ -11,7 +11,11 @@ const handler = async (req, res, client, getCurrentRound) => {
   } else if (segs[0] === 'retrievals' && req.method === 'PATCH') {
     await setRetrievalResult(req, res, client, Number(segs[1]), getCurrentRound)
```
Comment (juliangruber) [inline index.js:12]: Can we remove this route completely? SPARK clients have to update anyway and aren't rewarded yet, so I think we can prioritize the code cleanup now already

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: test/test.js (hunk 2 above)
Comment (bajtos) [reply test/test.js:364]: Sorry, that was a typo. I meant:
_After my change, they can submit as many measurements as they like._

File: index.js (hunk 3 above)
Comment (juliangruber) [reply index.js:12]: Got it, ok!

Comment (juliangruber) [inline index.js:12]:
```suggestion
    // TODO: Deprecate once clients have been updated
    await setRetrievalResult(req, res, client, Number(segs[1]), getCurrentRound)
```

File: test/test.js (hunk 2 above)
Comment (juliangruber) [reply test/test.js:364]: I think it's ok. We can discuss multiple strategies once we are more concerned with production deployment topics, or once we have many clients

**Review by juliangruber — APPROVED** [review]

**Pull request merged**